src/h5web/providers/hsds/api.ts

```diff
@@ -159,25 +176,6 @@ export class HsdsApi implements ProviderAPI {
     };
   }
 
-  /* Others */
-  public async getMetadata(): Promise<HDF5Metadata> {
-    if (this.metadata) {
-      return this.metadata;
-    }
-
-    const rootId = await this.fetchRoot();
-    await this.processGroup(rootId);
-
-    this.metadata = {
-      root: rootId,
-      groups: this.groups,
-      datasets: this.datasets,
-      datatypes: this.datatypes,
-    };
-
-    return this.metadata;
```

Comment:
We no longer need to store the metadata in the class, since `getMetadata` is only called once when the app renders (or when the provider's API changes, but same thing) and is not exposed to the app itself.